src/dom.js

```diff
@@ -161,6 +161,39 @@ export function createElementWithAttributes(doc, tagName, attributes) {
   return addAttributesToElement(element, attributes);
 }
 
+/**
+ * Returns true if node is connected (attached).
+ * @param {!Node} node
+ * @return {boolean}
+ * @see https://dom.spec.whatwg.org/#connected
+ */
+export function isConnectedNode(node) {
+  // "An element is connected if its shadow-including root is a document."
+  let n = node;
+  do {
+    n = rootNodeFor(n);
+    if (n.host) {
+      n = n.host;
+    } else {
+      break;
+    }
+  } while (true);
+  return (n instanceof Document);
```

Comment:
@jridgewell wow that is interesting! could you share if you have any reading material on this